src/pages/EnablePayments/PersonalInfo/PersonalInfo.tsx

```diff
@@ -0,0 +1,133 @@
+    const submit = () => {
+        // TODO: uncomment after connecting steps https://github.com/Expensify/App/issues/36648
+        // const personalDetails = {
+        //     phoneNumber: (values.phoneNumber && parsePhoneNumber(values.phoneNumber, {regionCode: CONST.COUNTRY.US}).number?.significant) ?? '',
+        //     legalFirstName: values?.[PERSONAL_INFO_STEP_KEYS.FIRST_NAME] ?? '',
+        //     legalLastName: values?.[PERSONAL_INFO_STEP_KEYS.LAST_NAME] ?? '',
+        //     addressStreet: values?.[PERSONAL_INFO_STEP_KEYS.STREET] ?? '',
+        //     addressCity: values?.[PERSONAL_INFO_STEP_KEYS.CITY] ?? '',
+        //     addressState: values?.[PERSONAL_INFO_STEP_KEYS.STATE] ?? '',
+        //     addressZip: values?.[PERSONAL_INFO_STEP_KEYS.ZIP_CODE] ?? '',
+        //     dob: values?.[PERSONAL_INFO_STEP_KEYS.DOB] ?? '',
+        //     ssn: values?.[PERSONAL_INFO_STEP_KEYS.SSN_LAST_4] ?? '',
+        // };
+        // Attempt to set the personal details
+        // Wallet.updatePersonalDetails(personalDetails);
+        Navigation.goBack(ROUTES.SETTINGS_WALLET);
+        Wallet.resetWalletAdditionalDetailsDraft();
+    };
+
+    const startFrom = useMemo(() => getInitialSubstepForPersonalInfo(values), [values]);
+
+    const {
+        componentToRender: SubStep,
+        isEditing,
+        nextScreen,
+        prevScreen,
+        moveTo,
+        screenIndex,
+        goToTheLastStep,
+    } = useSubStep({
+        bodyContent,
+        startFrom,
+        onFinished: submit,
+    });
+
+    const handleBackButtonPress = () => {
+        // TODO: connect to the fist step of the wallet setup https://github.com/Expensify/App/issues/36648
+        if (isEditing) {
+            goToTheLastStep();
+            return;
+        }
+        if (screenIndex === 0) {
+            Navigation.goBack(ROUTES.SETTINGS_WALLET);
+            Wallet.resetWalletAdditionalDetailsDraft();
+            return;
+        }
+        prevScreen();
+    };
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            testID={PersonalInfoPage.displayName}
+        >
+            <HeaderWithBackButton
+                title={translate('personalInfoStep.personalInfo')}
+                onBackButtonPress={handleBackButtonPress}
+            />
+            <View style={[styles.ph5, styles.mb5, styles.mt3, {height: CONST.BANK_ACCOUNT.STEPS_HEADER_HEIGHT}]}>
+                <InteractiveStepSubHeader
+                    startStepIndex={1}
+                    stepNames={CONST.WALLET.STEP_NAMES}
+                />
+            </View>
+            <SubStep
+                isEditing={isEditing}
+                onNext={nextScreen}
+                onMove={moveTo}
+            />
+        </ScreenWrapper>
+    );
+}
+
+PersonalInfoPage.displayName = 'PersonalInfoPage';
+
+export default withOnyx<PersonalInfoPageProps, PersonalInfoPageOnyxProps>({
```

Comment:
@Pujan92 The form didn't rerender when the data came from the hook - after refresh should show the Confirmation step when all the fields are filled in instead of the First and Last Name. It worked before I switched to using the hook, so I restored withOnyx. Probably some changes in useSubStep could solve it, but as it's also used in the VBBA flow I don't want to change it now - I'll do it with the other changes I plan to do when the flow is complete.